tests/ui/async-await/async-closures/def-path.stderr

```diff
@@ -0,0 +1,16 @@
+error[E0308]: mismatched types
+  --> $DIR/def-path.rs:8:9
+   |
+LL |     let x = async || {};
+   |                      -- the expected `async` closure body
+LL |     let () = x();
+   |         ^^   --- this expression has type `{static main::{closure#0}::{closure#0} upvar_tys=?7t witness=?8t}`
```

Comment:
`{closure#0}::{closure#0}` is proof of this def id relationship.

Reply:
Should we expect that proof in the test file? As a `//~| NOTE main::{closure#0}::{closure#0}` for instance.
r=me either way